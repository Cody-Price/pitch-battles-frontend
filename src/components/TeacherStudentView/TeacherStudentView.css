.teacher-student-view {
  font-family: "Open Sans", sans-serif;
  position: relative;
}
.teacher-student-view-header {
  width: 400px;
  margin: auto;
  display: flex;
  justify-content: space-between;
}
.teacher-student-view-header h2,
.teacher-student-view-header h3 {
  margin: 1rem;
}

.teacher-student-view-header p {
  font-size: 1.3rem;
}

.ts-total-games {
  font-weight: 700;
  margin-bottom: 0.3rem;
}

.teacher-student-view-total-games {
  margin-top: 0;
}

.teacher-student-view-times {
  margin-bottom: 2rem;
}
.teacher-student-perfect-row {
  display: flex;
  width: 400px;
  justify-content: space-between;
  margin: auto;
}
.perfect-score.true {
  background: url("./images/checkmark.png");
  background-size: cover;
  height: 32px;
  width: 32px;
  margin: auto;
}
.perfect-score.false {
  background: url("./images/x-symbol.png");
  background-size: cover;
  height: 32px;
  width: 32px;
  margin: auto;
}

.teacher-student-times-row {
  display: flex;
  width: 400px;
  justify-content: space-between;
  margin: auto;
}

.teacher-student-view-perfect-scores {
  padding-top: 1rem;
  border-top: 2px solid #333;
  width: 900px;
  margin: 0 auto 2rem;
}

.teacher-student-view-times h3,
.teacher-student-view-perfect-scores h3 {
  margin-bottom: 0;
}

.row-topper {
  border-top: 2px solid #333;
  padding-top: 1rem;
}

.game-id-label {
  width: 19px;
  margin-left: 1.1rem;
  margin-right: 2.1rem;
  font-weight: 700;
}

.game-perfect-mark-label {
  width: 137px;
  margin-left: 0;
  margin-right: 0;
  font-weight: 700;
}

.game-score-label {
  font-weight: 700;
}

.teacher-student-view-back-btn {
  position: absolute;
  right: 2rem;
  top: 1rem;
}
