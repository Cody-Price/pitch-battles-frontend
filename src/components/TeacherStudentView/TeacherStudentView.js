import React, { Component } from "react";
import "./TeacherStudentView.css";

import GameRow from "../GameRow/GameRow";

import { teacherBadgeHelper } from "../../utilities/teacherBadgeHelper";

class TeacherStudentView extends Component {
  constructor() {
    super();
    this.state = {
      perfectScores: []
    };
  }

  componentDidMount() {
    this.findPerfects();
  }

  findPerfects = () => {
    const perfectScores = teacherBadgeHelper(
      this.props.student.attributes.badges.data
    );

    this.setState({ perfectScores });
  };

  render() {
    const { attributes } = this.props.student;

    const perfectScoreRow = this.state.perfectScores.map(perfectScore => {
      return (
        <div
          key={perfectScore.display}
          className="teacher-student-perfect-block"
        >
          <p>{perfectScore.display}</p>
          <div className={`${perfectScore.earned} perfect-score`} />
        </div>
      );
    });

    const gameRows = this.props.student.attributes.games.data.map(game => {
      return <GameRow game={game} />;
    });

    return (
      <section className="teacher-student-view">
        <header className="teacher-student-view-header">
          <button
            className="teacher-student-view-back-btn"
            onClick={this.props.navigate}
          >
            back
          </button>
          <section className="header-left-wrapper">
            <h2 className="teacher-student-view-name">
              {attributes.first_name} {attributes.last_name}
            </h2>
            <h3 className="teacher-student-view-class-name">
              {attributes.class.data.attributes.name}
            </h3>
          </section>
          <div className="teacher-student-view-total-games-container">
            <p className="ts-total-games">Total Games</p>
            <p className="teacher-student-view-total-games">
              {attributes.total_games_played}
            </p>
          </div>
        </header>
        <section className="teacher-student-view-times">
          <h3>Fastest Times</h3>
          <section className="teacher-student-times-row">
            <div className="teacher-student-time-block">
              <p>Level One</p>
              <p>{attributes.level_one_fastest_time / 1000}</p>
            </div>
            <div className="teacher-student-time-block">
              <p>Level Two</p>
              <p>{attributes.level_two_fastest_time / 1000}</p>
            </div>
            <div className="teacher-student-time-block">
              <p>Level Three</p>
              <p>{attributes.level_three_fastest_time / 1000}</p>
            </div>
            <div className="teacher-student-time-block">
              <p>Level Four</p>
              <p>{attributes.level_four_fastest_time / 1000}</p>
            </div>
          </section>
        </section>
        <section className="teacher-student-view-perfect-scores">
          <h3>Perfect Scores</h3>
          <section className="teacher-student-perfect-row">
            {perfectScoreRow}
          </section>
        </section>
        <section className="game-row row-topper">
          <p className="game-id-label">id</p>
          <p className="game-score-label">L1 Time</p>
          <p className="game-perfect-mark-label">L1 Perfect</p>
          <p className="game-score-label">L2 Time</p>
          <p className="game-perfect-mark-label">L2 Perfect</p>
          <p className="game-score-label">L3 Time</p>
          <p className="game-perfect-mark-label">L3 Perfect</p>
          <p className="game-score-label">L4 Time</p>
          <p className="game-perfect-mark-label">L4 Perfect</p>
        </section>
        <section className="teacher-student-game-scroll">{gameRows}</section>
      </section>
    );
  }
}

export default TeacherStudentView;
