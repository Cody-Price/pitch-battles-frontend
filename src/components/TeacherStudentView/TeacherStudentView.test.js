import React from 'react';
import ReactDOM from 'react-dom';
import TeacherStudentView from './TeacherStudentView.js';
import Enzyme, { shallow } from 'enzyme';
const EnzymeAdapter = require('enzyme-adapter-react-16');

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('TeacherStudentView', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<TeacherStudentView />)
	});

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	})
})