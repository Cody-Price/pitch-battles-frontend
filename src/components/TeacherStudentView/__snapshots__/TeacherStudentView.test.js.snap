// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherStudentView should match the snapshot 1`] = `undefined`;
